### PR TITLE
Enables passing a frequency of 0 to an optimizer

### PR DESCRIPTION
**Pull request by asafmanor**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
**Enables passing a frequency of 0 to an optimizer (valid option when you want to "turn off" one optimizer). Before the change it would raise the ValueError.**

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
just curious what is the motivation to interpret 0 as frequency 1?

**Comment by jeremyjordan** [comment]
@asafmanor can you explain a scenario under which you would want to define an optimizer but never run it? that's how i'm interpreting this change.

**Comment by asafmanor** [comment]
Yes, sure.
I am now conducting experiments with a GAN, and in my ablation study, I run
some experiments with no adversarial loss.
A very simple way to do that would be to pass a frequency of 0.

Another option - when finetuning a network, you might want to reduce to
frequency to 0 in the second stage after you have finished training one of
its components.
When I research, I always stumble upon these unusual scenarios.

**Comment by justusschock** [comment]
I'm not sure, I like this. IMO when you specify an optimizer, it should also have a frequency >= 0. Everything else does not feel intuitive for me. For such cases I think the cleanest approach would be to create a child class and overwrite your optimizer specs

**Comment by williamFalcon** [comment]
i think @asafmanor is saying 0 is not a valid frequency? @justusschock you suggest >=0 which makes sense but i think we only support > 0 and this PR enables >=0 support?

**Comment by asafmanor** [comment]
As I understand I have the minority opinion that 0 **is** a valid frequency and @justusschock believes it isn't.

**+I'll bump into this thread with another related issue:**
I've noticed is that when running a`fast_dev_run`only one training step is done, thus not all optimizers are called and not all the different parts of the `training_step()` are checked.
Is that true? should we fix it?

**Comment by williamFalcon** [comment]
1. please rebase
2. i see both sides here. i do generally err in increasing flexibility for edge-case research cases without adding more ways of doing one thing. with that said, i think supporting freq of zero is not a dealbreaker and likely useful in some research settings. 

i guess i’m interpreting this change in that you want to define both optimizers and sometimes not run both (maybe at the end of training where one has converged, then you want to skip this one going forward. 

i assume if you actually never wanted to run one optimizer of the two, you would just use one haha. 

if the above is true, then i’m happy to support this PR. 

3. fast_dev_run should run both optimizers. mind submitting a separate PR?

**Comment by asafmanor** [comment]
Thanks!
3. Yeah, I'll try to do that this week.
